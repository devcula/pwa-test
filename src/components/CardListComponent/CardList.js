import React from 'react';
import Card from '../CardComponent/Card';

const CardList = ( {robots} ) =>{
    if(robots.length===0){
        return (
            <div>
                <h1>Nothing found to display</h1>
            </div>
        )
    }
    else{
        return(
            <div>
                {
                    robots.map( (user,i) =>{
                        return (
                        <Card 
                        key={i} 
                        id = {user.id} 
                        name={user.name} 
                        email={user.email}
                        />)
                    })
                }
            </div>
        )
    }
}

export default CardList;