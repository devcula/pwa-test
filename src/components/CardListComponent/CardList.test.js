import { shallow } from 'enzyme';
import CardList from './CardList';
import React from 'react';
import toJson from 'enzyme-to-json';

const mockRobots = [
    {
        id: 1,
        name: "John",
        username: "JohnJohn",
        email: "John@gmail.com"
    }
]

it("expect to render CardList component with 1 robot", () => {
    const wrapper = shallow(<CardList robots={mockRobots} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(toJson(wrapper).children.length).toEqual(1);
})

it("expect to render CardList component with an empty list", () =>{
    const wrapper = shallow(<CardList robots={[]} />);
    expect(toJson(wrapper).children.length).toEqual(1);
})