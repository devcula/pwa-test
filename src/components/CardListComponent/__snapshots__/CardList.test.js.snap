// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList component with 1 robot 1`] = `
<div>
  <Card
    email="John@gmail.com"
    id={1}
    key="0"
    name="John"
  />
</div>
`;
